src/karmen_frontend/cypress/integration/gcodes/listing_spec.js

```diff
@@ -0,0 +1,141 @@
+import { Chance } from "chance";
+const chance = new Chance();
+
+const visitGCodes = () => {
+  cy.get('button[id="navigation-menu-toggle"]').click();
+  return cy.get('a[id="navigation-gcodes"]').click();
+};
+
+describe("G-codes: Listing - no G-codes uploaded", function () {
+  let email, password, organizationUuid;
+  beforeEach(() => {
+    email = chance.email();
+    password = chance.string();
+    return cy
+      .logout()
+      .createUser(email, password)
+      .login(email, password)
+      .then((data) => {
+        organizationUuid = Object.keys(data.organizations)[0];
+      })
+      .then(() => {
+        cy.logout()
+          .login(email, password)
+          .then(() => {
+            return visitGCodes();
+          });
+      });
+  });
+
+  it("search", function () {
+    cy.get(".list-item").findByText("No items found!").should("exist");
+
+    cy.get("#filter").type("Non existing Gcode");
+    cy.get(".list-item").findByText("Non existing Gcode").should("not.exist");
+    cy.get(".list-item").findByText("No items found!").should("exist");
+  });
+});
+
+describe("G-codes: Listing", function () {
+  let email, password, organizationUuid;
+  beforeEach(() => {
+    email = chance.email();
+    password = chance.string();
+    return cy
+      .logout()
```

Comment:
This might better be placed in some global `beforeEach`..

Reply:
Added a new command for setting new logged user.

Reply:
Neat.